### PR TITLE
[Merged by Bors] - feat(CategoryTheory/Limits/ConcreteCategory): prove some results for concrete category with extra algebraic structures

### PR DESCRIPTION
**Pull request by jjaassoonn**
---

These results are necessary in #14809, but we don't want to change the import too much.

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/CategoryTheory/Limits/ConcreteCategory/WithAlgebraicStructures.lean [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+import Mathlib.CategoryTheory.Limits.ConcreteCategory.Basic
+import Mathlib.Algebra.Module.LinearMap.Defs
+import Mathlib.Tactic.CategoryTheory.Elementwise
+
+/-!
+
+# Colimits in concrete categories with algebraic structures
+
+Let `C` be a concrete category and `F : J ⥤ C` a diagram in `C`. We discuss some results about
+`colimit F` when objects and morphisms in `C` has some algebraic structures.
```
Comment (dagurtomas) [inline Mathlib/CategoryTheory/Limits/ConcreteCategory/WithAlgebraicStructures.lean:15]:
```suggestion
Let `C` be a concrete category and `F : J ⥤ C` a filtered diagram in `C`. We discuss some results about
`colimit F` when objects and morphisms in `C` have some algebraic structures.
```

File: Mathlib/CategoryTheory/Limits/ConcreteCategory/WithAlgebraicStructures.lean [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+## Main results
+- `CategoryTheory.Limits.Concrete.colimit_rep_eq_zero`: Let `C` be a category where its objects has
+  zero elements and morphisms preserves zero. If `x : Fⱼ` is mapped to `0` in the colimit, then
+  there exists a `i ⟶ j` such that `x` restricted to `i` is already `0`.
```
Comment (dagurtomas) [inline Mathlib/CategoryTheory/Limits/ConcreteCategory/WithAlgebraicStructures.lean:20]:
```suggestion
- `CategoryTheory.Limits.Concrete.colimit_rep_eq_zero`: Let `C` be a category where its objects have
  zero elements and morphisms preserve zero. If `x : Fⱼ` is mapped to `0` in the colimit, then
  there exists a `i ⟶ j` such that `x` restricted to `i` is already `0`.
```

**Review by dagurtomas — APPROVED** [review]

**Comment by dagurtomas** [comment]
maintainer merge

**Comment by riccardobrasca** [comment]
Thanks!

bors merge

**Pull request closed without merging**